.buildkite/scripts/run_e2e_tests.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -euo pipefail
+
+VERSION_URL="https://storage.googleapis.com/artifacts-api/releases/current"
+
+###
+# Resolve stack version and export
+resolve_current_stack_version() {
+  set +o nounset
+
+  local major_version="${ELASTIC_STACK_VERSION%%.*}"
+  local version=$(curl --retry 5 --retry-delay 5 -fsSL "$VERSION_URL/$major_version")
+
+  echo "Resolved version: $version"
+  export STACK_VERSION="$version"
+}
+
+resolve_current_stack_version
+
+###
+# Build the plugin, to do so we need Logstash source
+build_logstash() {
+  retry -t 5 -- git clone https://github.com/elastic/logstash.git
```

Comment:
Our base images have the logstash repo precloned so that we can use the [reference repo](https://git-scm.com/docs/git-clone) feature to speed up cloning and be nice to GitHub wrt bandwidth usage.

What we'd need to do here is something like:

`retry -t 5 -- git clone -v --reference /usr/local/git-references/git-github-com-elastic-logstash-git -- https://github.com/elastic/logstash.git .`

Bonus points if we check first (to allow running this script locally, during development) i.e.

```
if [[ -d /usr/local/git-references/git-github-com-elastic-logstash-git ]]; then
    retry -t 5 -- git clone -v --reference /usr/local/git-references/git-github-com-elastic-logstash-git -- https://github.com/elastic/logstash.git .
else
    retry -t 5 -- git clone https://github.com/elastic/logstash.git
fi
```

Reply:
Good point! Will `--reference` be useful especially on retries or some agents already cloned (couldn't figure out its benefits for our case)?
I have also added `--single-branch` option which I believe it will also somehow contributes to better performance or avoids unnecessary (sor far) fetchings.

Reply:
> Will --reference be useful especially on retries or some agents already cloned (couldn't figure out its benefits for our case)?

Reference repos are a local bare repository whose content is used instead of copying from the remote repository.
If we don't use them, we are unnecessarily downloading a lot of data which is time consuming, error prone and not very kind to GitHub :)

Since the VM images get rebuilt everything week, the content of the logstash reference will have very little drift from the latest HEAD, therefore help in all above mentioned ways.